bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -164,51 +193,68 @@ pub mod pallet {
 			let asset_location: Location =
 				(*asset_id).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
 
-			let mut checked = false;
-			if asset_location.eq(&origin_location) || asset_location.starts_with(&origin_location) {
-				checked = true
-			}
-			ensure!(checked, <Error<T>>::OwnerCheck);
+			// Burn Ether Fee for the cost on ethereum
+			Self::burn_for_teleport(&origin_location, &(T::FeeAsset::get(), fee).into())?;
+
+			// Burn RemoteExecutionFee for the cost on bridge hub
+			Self::burn_for_teleport(&origin_location, &T::RemoteExecutionFee::get())?;
 
-			Self::burn_fees(origin_location.clone(), fee)?;
+			let reanchored_asset_location = asset_location
+				.clone()
+				.reanchored(&T::BridgeHubLocation::get(), &T::UniversalLocation::get())
+				.map_err(|_| Error::<T>::LocationConversionFailed)?;
 
-			let call = SnowbridgeControl::Control(ControlCall::RegisterToken {
-				asset_id: Box::new(VersionedLocation::from(asset_location.clone())),
+			let call = BridgeHubRuntime::EthereumSystem(EthereumSystemCall::RegisterToken {
+				asset_id: Box::new(VersionedLocation::from(reanchored_asset_location.clone())),
 				metadata,
 				fee,
 			});
 
+			let topic = blake2_256(&("SnowbridgeFrontend", call.clone()).encode());
+
 			let xcm: Xcm<()> = vec![
```

Comment:
As with the inbound-queue, you should descend the origin to the pallet of the `system-frontend` pallet on AH.

https://github.com/vgeddes/polkadot-sdk/blob/19f7f03635861fc31c6f5a9b2123585e44a2153a/bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs#L200

https://github.com/vgeddes/polkadot-sdk/blob/19f7f03635861fc31c6f5a9b2123585e44a2153a/bridges/snowbridge/pallets/inbound-queue-v2/src/mock.rs#L172